src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -0,0 +1,947 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides an implementation of the BLEManager singleton object
+ *          for the ESP32 (NimBLE) platform.
+ */
+/* this file behaves like a config.h, comes first */
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+#if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
```

Comment:
What are the thoughts about removing the entire file from compilation from a compile setting rather than checking #ifdefs like this on the entire file?

Reply:
We need to decide which model we'd like to follow.  I had proposed this pattern of #ifdef wrapped sources, which is what OpenThread uses, so we avoid build system-specific dependencies and can configure the entire project with a simple config.h.

Reply:
Hi @andy31415 @turon sorry for delayed response, I have used "#ifdef" with this project. Please let me know if you have any better alternatives in mind.